### DISCUSSION TITLE
Getting error using the Select component

### DISCUSSION THREAD

**Discussion by devstc** [mantinedev/mantine#1815] [Q&A] [answered]
This may not be something Mantine related.  It could easily be my lack of coding experience related.  But I wanted to check here first, and if I need to seek answers elsewhere then I will.  I won't tell you how long I'm been trying to solve this, but I can promise I have spent a considerable amount of time on Google before coming here.

As to my issue... I've got a functional component that creates a form that allows a user to add a new project to a database.  The form has 7 fields, but the code here shows just one of the (Mantine) _Select_ components which is where I'm having the difficulty.  The field on the form allows the user to select from a list of Project Manager names that have been pulled from a database, put into an array, and then that array is given to the _data_ prop of the _Select_ component.

When the form is displayed (in a Mantine _Modal_), when I click the form field and then click on a name from the available list, I get the following error thrown (the line of code it's referring to is the fourth line of the handleChange function just above the _return_ block):
```
Uncaught TypeError: Cannot read properties of undefined (reading 'name') at AddProject.tsx:39:1
```

Is there something related to how the _Select_ component works that means I have to supply the data to the _data_ prop differently that how I am?  Previously, I have also given the _data_ prop the data as a more basic array like this: ["name", "name", "name"], rather than creating an object for each name with a **label, name,** and **value** property, but I had exactly the same problem that way too.  Like I said at the beginning, this may not be something Mantine-related, but I wanted to ask here first so I could mark that off the list if that isn't where the issue lies.

```
export default function AddProject(props: any) {

   const users = props.users;

   interface projMgr {
      label: string
      name: string
      value: string
   }

   const [pMgrNames, setpMgrNames] = useState<projMgr[]>([]);

   useEffect(() => {
      makePMgrNamesArr()
   }, [users])

   const makePMgrNamesArr = async () => {
      const namesArr = await users.map((userx: projMgr) => {
         const labelVal = userx.name
         const nameVal = userx.name
         const valueVal = userx.name
         return {label: labelVal, name: nameVal, value: valueVal}
      })
      setpMgrNames(namesArr)
   }

   const [formData, setFormData] = useState({manager: ""})

   const {manager} = formData

   const handleChange = (e: any) => {
      setFormData((prevState) => ({
         ...prevState,
         [e.target.name]: e.target.value
      }))
   }

   return (
      <form onSubmit={handleSubmit}>
         <Select
            placeholder="<click to select>"
            label="Project Manager"
            data={pMgrNames}
            type="text"
            id="manager"
            name="manager"
            value={manager}
            onChange={handleChange}
         />
      </form>
   )
}
```

**Comment by rtivital** [comment]
Select component `onChange` handler does not use event, it is called with value you should change your `handleChange` handler to be:

```tsx
const handleChange = (value: string) => {
      setFormData((prevState) => ({
         ...prevState,
         manager: value
      }))
   }
```

**Comment by devstc** [comment]
Thanks Vitaly.  That works for making the one input type work.  I only included the one type of input in my post to try to help with clarity, but I'm also working with _TextInput_ components in the same form.  Do I need 2 separate `onChange` handlers, one for _Select_ components and one for _TextInput_ components?  Because in the documentation for the _TextInput_ component under the 'Controlled' heading, it shows the _TextInput_ component DOES use the event in the `onChange` handler, whereas as you explained, the _Select_ component does not.

**Reply by rtivital** [reply]
Yes, you need to have two separate handlers for that, or you can check payload type to be an event in your `handleChange` function. There are also alternatives like `use-form` and `use-input-state` that will handle this logic for you.

**Comment by devstc** [comment, marked as answer]
Thanks Vitaly, that was helpful.  Just knowing _TextInput_ and _Select_ need their own separate handlers got me away from trying to create a one-size-fits-all solution that had to work for both.  A little code wrestling and I was able to get my form working!  Maybe I'll look into the payload type checking you mentioned, but for now it's working, and that's all I need for now.  Thanks again!
